smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -258,95 +295,137 @@ def _on_iteration(self) -> None:
         timings.append(time.perf_counter() - interm)  # timing
         interm = time.perf_counter()  # timing
 
+        reply = InferenceReply()
+
         if not request.raw_model:
             if request.model_key is None:
-                # A valid request should never get here.
-                raise ValueError("Could not read model key")
+                send_failure(request.callback, "Could not find model key or model.")
+                return
             if request.model_key in self._cached_models:
                 timings.append(time.perf_counter() - interm)  # timing
                 interm = time.perf_counter()  # timing
                 model_result = LoadModelResult(self._cached_models[request.model_key])
 
             else:
-                fetch_model_result = None
-                while True:
-                    try:
-                        interm = time.perf_counter()  # timing
-                        fetch_model_result = self._worker.fetch_model(
-                            request, self._feature_store
-                        )
-                    except KeyError:
-                        time.sleep(0.1)
-                    else:
-                        break
-
-                if fetch_model_result is None:
-                    raise SmartSimError("Could not retrieve model from feature store")
+                try:
+                    interm = time.perf_counter()  # timing
```

Comment:
it seems funny that this time.perf_counter is inside the try block but the next one isn't. Should it be?

Reply:
I'll ask Al if these should all be inside the try blocks or outside of them so there's consistency.

Reply:
Asked Al and I made sure they're all outside of the try blocks.